pkg/clients/compute.go

```diff
@@ -154,6 +186,21 @@ func (c computeClient) ListServerGroups() ([]servergroups.ServerGroup, error) {
 	return servergroups.ExtractServerGroups(allPages)
 }
 
+// WithMicroversion checks that the required Nova microversion is supported and sets it for
+// the ComputeClient.
+func (c computeClient) WithMicroversion(required string) (ComputeClient, error) {
+	supported, err := openstackutil.MicroversionSupported(required, c.minVersion, c.maxVersion)
+	if err != nil {
+		return nil, err
+	}
+	if !supported {
+		return nil, fmt.Errorf("microversion %s not supported. Min=%s, max=%s", required, c.minVersion, c.maxVersion)
+	}
+	versionedClient := c
+	versionedClient.client.Microversion = required
```

Comment:
Because client is a pointer this is actually changing the Microversion globally, meaning we need to be careful to put it back again after. I wonder, is there any way to copy the ServiceClient? I haven't checked, but I have a feeling that creating a ServiceClient may not involve any API calls? If so, perhaps we could just create a new one?

Reply:
Oh I see! We should probably do a copy. Creating a new requires one API call to check supported microversions:
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/7c38ac82f70f8c78e0b0244db80775c4881be8ef/pkg/clients/compute.go#L91